packages/p2p/src/components/order-details/order-details.jsx

```diff
@@ -141,7 +143,10 @@ const OrderDetails = observer(() => {
             showModal({ key: 'EmailLinkExpiredModal' }, { should_stack_modal: isMobile() });
         }
 
-        if (status_string === 'Expired' && isCurrentModal('EmailLinkExpiredModal'))
+        if (
+            status_string === 'Expired' &&
+            (isCurrentModal('EmailLinkExpiredModal') || isCurrentModal('OrderDetailsConfirmModal'))
```

Comment:
The isCurrentModal function can actually take in an array of modals, could you try and check if this is ok?

```suggestion
(isCurrentModal('EmailLinkExpiredModal', 'OrderDetailsConfirmModal'))
```